scripts/pipeline_generator/plugin.py

```diff
@@ -0,0 +1,90 @@
+from pydantic import BaseModel, Field
+from typing import List, Dict, Any, Optional
+
+from .utils import HF_HOME
+
+DOCKER_PLUGIN_NAME = "docker#v5.2.0"
+KUBERNETES_PLUGIN_NAME = "kubernetes"
+
+class DockerPluginConfig(BaseModel):
+    image: str = ""
+    always_pull: bool = Field(default=True, alias="always-pull")
+    propagate_environment: bool = Field(default=True, alias="propagate-environment")
+    gpus: Optional[str] = "all"
+    mount_buildkite_agent: Optional[bool] = Field(default=False, alias="mount-buildkite-agent")
+    command: List[str] = Field(default_factory=list)
+    environment: List[str] = [
+        f"HF_HOME={HF_HOME}",
+        "VLLM_USAGE_SOURCE=ci-test",
+        "HF_TOKEN",
+        "BUILDKITE_ANALYTICS_TOKEN"
+    ]
+    volumes: List[str] = [
+        "/dev/shm:/dev/shm",
+        f"{HF_HOME}:{HF_HOME}"
+    ]
+
+class KubernetesPodContainerConfig(BaseModel):
```

Comment:
we cancelled the plan to move back to k8s stack? why are we adding this?

Reply:
This is for the A100 node on EKS (roblox aws)